zilliqa/src/contracts/mod.rs

```diff
@@ -39,220 +36,119 @@ pub mod native_token {
-
-        let solc_path = temp_dir.path().join("solc");
-        println!("{}", solc_path.to_string_lossy());
-        let mut solc_file = OpenOptions::new()
-            .mode(0o777)
-            .write(true)
-            .create(true)
-            .truncate(true)
-            .open(&solc_path)
-            .unwrap();
-        solc_file.write_all(&solc).unwrap();
-        mem::drop(solc_file); // Close the file.
-
-        let root = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
-            .join("src")
-            .join("contracts");
-        let contract = root.join(filename);
-        let vendor = root
-            .parent()
-            .unwrap()
-            .parent()
-            .unwrap()
-            .parent()
+    fn compile_all() {
+        let root = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        let input = CompilerInput {
+            language: "Solidity".to_owned(),
+            sources: Source::read_all(
+                [
+                    "gas_price.sol",
+                    "native_token.sol",
+                    "shard.sol",
+                    "shard_registry.sol",
+                ]
+                .map(|c| format!("src/contracts/{c}")),
+            )
+            .unwrap(),
+            settings: Settings {
+                remappings: vec![Remapping {
+                    context: None,
+                    name: "@openzeppelin".to_owned(),
+                    path: "../vendor/openzeppelin-contracts/".to_owned(),
+                }],
+                optimizer: Optimizer {
+                    enabled: Some(true),
+                    runs: Some(4294967295),
+                    details: None,
+                },
+                output_selection: OutputSelection::complete_output_selection(),
+                evm_version: Some(EvmVersion::Shanghai),
+                ..Default::default()
+            },
+        };
+
+        let solc = Solc::find_or_install_svm_version("0.8.23")
             .unwrap()
-            .join("vendor");
+            .with_base_path(&root)
+            .args(["--allow-paths", "../vendor"]);
 
-        let output = Command::new(solc_path)
-            .current_dir(root)
-            .arg(contract)
-            .arg("@openzeppelin/=openzeppelin-contracts/")
-            .arg("--base-path")
-            .arg(".")
-            .arg("--include-path")
-            .arg(vendor)
-            .arg("--combined-json")
-            .arg("abi,bin,bin-runtime")
-            .arg("--evm-version=paris")
-            .output()
-            .unwrap();
+        let output = solc.compile_exact(&input).unwrap();
+        let output_file = root.join("src").join("contracts").join("compiled.json");
 
-        eprintln!("{}", std::str::from_utf8(&output.stderr).unwrap());
-        let combined_json: Value = serde_json::from_slice(&output.stdout).unwrap();
-        let contract = combined_json["contracts"][&json_key].clone();
+        if std::env::var_os("ZQ_CONTRACT_TEST_BLESS").is_some() {
```

Comment:
Bless? What is it supposed to mean? I might suggest something like `...TEST_ONEOFF_CONTRACT_COMPILE` or something as it is non-obvious.

Reply:
I used "bless" to mean "bless the current output as the new expected output". I think I subconsciously stole the term from rustc: https://rustc-dev-guide.rust-lang.org/tests/ui.html#output-comparison

> UI tests store the expected output from the compiler in .stderr and .stdout files next to the test. You normally generate these files with the --bless CLI option, and then inspect them manually to verify they contain what you expect.